metadata-ingestion/src/datahub/ingestion/source/sql/postgres.py

```diff
@@ -67,10 +126,93 @@ class PostgresSource(SQLAlchemySource):
     - Table, row, and column statistics via optional SQL profiling
     """
 
-    def __init__(self, config, ctx):
+    def __init__(self, config: PostgresConfig, ctx: PipelineContext):
         super().__init__(config, ctx, "postgres")
 
     @classmethod
     def create(cls, config_dict, ctx):
         config = PostgresConfig.parse_obj(config_dict)
         return cls(config, ctx)
+
+    def get_workunits(self) -> Iterable[Union[MetadataWorkUnit, SqlWorkUnit]]:
+        yield from super().get_workunits()
+
+        if self.config.include_view_lineage:  # type: ignore
+            yield from self._get_view_lineage_workunits()
+
+    def _get_view_lineage_workunits(self) -> Iterable[MetadataWorkUnit]:
+        url = self.config.get_sql_alchemy_url()
+        engine = create_engine(url, **self.config.options)
+
+        data: List[ViewLineageEntry] = []
+
+        with engine.connect() as conn:
```

Comment:
this method is pretty long - could you refactor it into two, one which runs the query and creates the `lineage_elements` dict, and the second which turns those lineage elements into lineage MCEs?

Reply:
done